Void Finisher now survives the Void

AlinaStarkova opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
It makes it so when you finish the Void Path as a Heretic, you can actually survive in the void (as well as not suffocate in your own storm), you no longer need to breathe, and you resist low pressure (but not high pressure). (the add_traits show a reference to space_adaptation, it doesnt seem to work if i change the reference, and it doesnt seem to have any issues referencing that, so it should be fine as is)
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
I originally setout to fix a really severe issue with the Void finisher, namely that the cold air it produces brings the oxygen levels around the Heretic so low, that you would actually suffocate. How odd it is that you do all that work for the Void Path, just to suffocate with your own finisher. during the fixing of this issue (making it so the heretic who finishes the Void Path no longer needs to breathe), i felt it was also odd that the Void is still deadly to the Aristocrat of the Void.
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl: Nari Harimoto
tweak: Heretics who finish the Void Path and become an Aristocrat of the Void can now survive in the Void (space).
fix: Heretics who research Aristocrats Way on the Void Path will no longer suffocate in their own storm when they ascend (no longer breathes).
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

AlinaStarkova committed:
Void Finisher now survives the Void

it was pretty dumb that your VOID finisher would die in the void

ATH1909 commented on code/modules/antagonists/eldritch_cult/knowledge/void_lore.dm:
```diff
@@ -162,6 +162,8 @@
 	var/mob/living/carbon/human/H = user
 	H.physiology.brute_mod *= 0.5
 	H.physiology.burn_mod *= 0.5
+	ADD_TRAIT(H, TRAIT_RESISTLOWPRESSURE, "space_adaptation")
+	ADD_TRAIT(H, TRAIT_NOBREATH, "space_adaptation")
```
the source for this should be something other than "space_adaptation", or else gaining and losing the space adaptation mutation will also cause you to lose these traits

ATH1909 commented:
I'd personally put these traits in the Aristocrat's Way tech, to make it a bit less underwhelming (it currently ONLY provides cold immunity). Because all void heretics have to take the Aristocrat's Way tech before they can branch out to other paths/continue on down the void path, and only void heretics can take the void ascension tech, there's no way that you can get the Void Ascension tech without getting the Aristocrat's Way tech (although you could theoretically get debrained and put in a new body between researching Aristocrat's Way and finishing the Void Ascension ritual...).

AlinaStarkova committed:
Update code/modules/antagonists/eldritch_cult/knowledge/void_lore.dm

Co-authored-by: ATH1909 <42606352+ATH1909@users.noreply.github.com>

AlinaStarkova commented:
> 
> 
> I'd personally put these traits in the Aristocrat's Way tech, to make it a bit less underwhelming (it currently ONLY provides cold immunity). Because all void heretics have to take the Aristocrat's Way tech before they can branch out to other paths/continue on down the void path, and only void heretics can take the void ascension tech, there's no way that you can get the Void Ascension tech without getting the Aristocrat's Way tech (although you could theoretically get debrained and put in a new body between researching Aristocrat's Way and finishing the Void Ascension ritual...).

thats fine, thanks, it just wasnt working if i tried to change the path away from space adaptation, and im brand new to all these files so it was hard finding something that would work

AlinaStarkova committed:
Changes ability to not need to breathe to Aristocrats Way

you no longer need to breathe earlier in the Void Path, you can replicate this simply by wearing internals, however cold immunity isnt true immunity when the cold also shrinks the o2 in the air, making it not able to breathe, this makes it so you dont need to worry about breathing; leaves low pressure immunity for the finisher

AlinaStarkova committed:
Update Aristrocrats WAy description

as it says on the tin

AlinaStarkova commented:
people have suggested adding the lowpressure immunity to aristocrats way as well, but i feel thats too early a tech to have nobreath and void immunity together, those are very strong together, plus the aristocrats way description talks about cold breath only. if people really want i can move it up, but i feel until a void heretic truely ascends they arent fully immune to the void, but i can maybe add a half off damage modifer for low pressure on aristrocrats way if requested

Borisvanmemes commented:
You can simply grab an emergency firesuit for pressure immunity and spacewalk that way, it really wouldn't be that big of a buff.

AlinaStarkova commented:
> 
> 
> You can simply grab an emergency firesuit for pressure immunity and spacewalk that way, it really wouldn't be that big of a buff.

emergency firesuits arent pressure immune, just pressure resistant, and while not a big buff, the point of this isnt really to be a buff, rather 1. the nobreathe fixes the issue of suffocating in your own storm (yes yes, you can wear internals.... not the point) as well as fix the lore issues surrounding being a VOID aristocrat which literally dies in the void

there are other buffs that CAN be made, but this is more of a tweak/fix rather than an outright buff, wanna see a buff? checkout ash ascension, you get nobreathe, low AND high pressure immunity, hot AND cold (why cold???) immunity, fire immunity... and half off damage modifer for brute and burn!! talk about overpowered, currently void has..... nothing, just a cool storm and half off brute and burn, thats IT. this will hopefully be a small fix that atleast gives void SOMETHING worthwhile aside from cool music and a storm that kills its own heretic kek

EdgeLordExe commented:
good change, i actually forgot to add resistances to void ascension.

Borisvanmemes commented:
Emergency Firesuits are flagged STOPSPRESSUREDAMAGE, same as spacesuits.
If you chug a load of coffee and have air you can safely spacewalk with them.

AlinaStarkova commented:
> Emergency Firesuits are flagged STOPSPRESSUREDAMAGE, same as spacesuits.
> If you chug a load of coffee and have air you can safely spacewalk with them.

they stop pressure damage to the parts they cover, ie. your chest, your head is still exposed and hence you still take SOME damage, just not as much as you would normally

regardless, going "b-but firesuits exist!" shouldn't be a reason to give a much needed tweak to how void heretics function. even the coder who made the heretic antag admitted here on this PR that he forgot to add resistances to void. this is a fix.

AlinaStarkova commented:
> A Firefighter Helmet finishes the set off and allows full pressure immunity, I apologize for not mentioning that.
> If you only wanna fix a bug in this PR it ain't no skin off my back.

the firefighter helmet is a hat, not a helmet, it doesnt provide full pressure immunity

by your logic, lets remove hardsuits and eva suits, afterall emergency firesuits are good enough right?

Borisvanmemes commented:
I'm sorry but what the hell are you going on about?

AlinaStarkova commented:
Hmmmmm, this PR is done for what it set out to accomplish, one can only wait it seems

LemonInTheDark commented on code/modules/antagonists/eldritch_cult/knowledge/void_lore.dm:
```diff
@@ -47,7 +47,7 @@
 /datum/eldritch_knowledge/cold_snap
 	name = "Aristocrat's Way"
-	desc = "Makes you immune to cold temperatures, you can still take damage from lack of pressure."
+	desc = "Makes you immune to cold temperatures, and you no longer need to breathe, you can still take damage from lack of pressure."
```
Aren't you changing this last bit so they don't take damage from lack of pressure/I kinda get the pressure thing, but don't all crewmembers spawn with air?

AlinaStarkova replied:
for the same reason the ash heretic gets nobreath, cold immunity, lowpressure immunity, high pressure immunity, hot immunity, as well as damage resistances. currently void heretic has zero perks/traits (guy who coded it said he forgot). so it may not be as good as the other paths perks, but its still something, and i mainly noticed it as an issue when the heretic spawns the void storm but wears a mask that cant use internals (the masks the heretics can get cant use internals) ... the thing with void storms is the freeze the air, lowering the o2 pressure, so if you dont have internals you will suffocate.... inside your very own storm. so its just one extra bonus that fits thematically with with the ascension path of void (why would a void aristrocrat need to breathe when he rules the void, aka an airless space?)

if anything the other paths need some traits removed, ash path for example.... why exactly is a fire heretic immune to cold? it doesnt really fit thematically but im focusing on the void path with this

LemonInTheDark replied:
Do they still take damage from low pressure? because it looks like you gave them resistance to it.

AlinaStarkova replied:
they do not, the finisher makes them immune from low pressure, seeing as they are the Aristocrat of the Void, it has been tested, it is full immunity, not a resistance. I have spoken with the guy who made heretics, he intended for Void Heretics to have immunities and stuff, he just forgot. even with just these two changes, Void heretic has not even 30% of the immunities that the other heretics get as their finisher, so i dont feel this to be particularly unbalanced, relatively speaking

https://user-images.githubusercontent.com/40489693/102828346-83d5d200-43b2-11eb-99a5-a92d43faf046.mp4

LemonInTheDark replied:
You should update the text properly then, because it still talks about taking damage from low pressure

AlinaStarkova replied:
no no, you misunderstand, the text your reviewing here is for the nobreathe, this is on the arisocrats way tech which is one of the first techs on the void path. you DO still take damage from low pressure at that point, full low pressure immunity only comes when you ascend.
